### ISSUE TITLE
Using deprecated attrdict library failing with python 3.10

### ISSUE THREAD

**Issue by SOMO-Prevas** [wallento/wavedrompy#32]
The attrdict library is deprecated and use "from collections import Mapping", etc. which has been deprecated in python since 3.3 and is not supported in python 3.10.

**Comment by MattEttus** [comment]
I believe this can be fixed with this -- 

https://stackoverflow.com/questions/69381312/in-vs-code-importerror-cannot-import-name-mapping-from-collections

**Comment by bavovanachte** [comment]
@wallento I'm getting an increasing amount of reports on this from sphinx plugin users. By looking at the overall activity on the project I understand it isn't actively maintained anymore, but since many people still seem to depend upon it when using it from readthedocs, could the python 3.10 still be supported?

Also, is there someone who could take over maintenance of the project?

**Comment by wallento** [comment]
Hi,

I am happy to pick it up again. Sorry, it seemed to me it was stable and of decreasing use due to the rise of the npm-based version using original wavedrom. Will pick that up asap, but also happy to hand over maintenance to anyone interested.

Cheers

**Comment by bavovanachte** [comment]
> Sorry, it seemed to me it was stable and of decreasing use due to the rise of the npm-based version using original wavedrom. 

From what I can gather, this is still very useful for rtd users who don't have the option to install npm packages. Not sure if that's going to change anytime soon (have never used the service myself, so don't know for sure) 

By definition this project will always be behind the officially maintained tool, and I don't think anyone should expect it to always be up to date immediately. So as long as it's not crashing and it's clear what version of upstream wavedrom you're compatible with (I like the idea of https://github.com/wallento/wavedrompy/pull/26/files) I think most users will be happy :)

**Comment by wallento** [comment]
Unfortunately there was no interest from upstream wavedrom to coordinate on updates and make them easier to follow, which made it a major occasional reverse engineering effort. On the other hand those major update dumps were rather occasional and I can try to set a day aside to catch up with another version, at least up to picked features that were contributed (like the colors).

**Comment by wallento** [comment]
The problem of this merge is solved on master, I will have a look at CI to test if it works and look forward to confirmation too.

**Comment by Araneidae** [comment]
Forgive me, but why is this closed without a corresponding release?  I'm currently having to pin my pip install to work around issue https://github.com/bavovanachte/sphinx-wavedrom/issues/38; I'd be grateful for a release, please!

**Comment by ryananguiano** [comment]
You can monkeypatch the collections module near the start of your app before the module is imported elsewhere.

```python
# Monkey patch collections
import collections
import collections.abc
for type_name in collections.abc.__all__:
    setattr(collections, type_name, getattr(collections.abc, type_name))
```

**Comment by wallento** [comment]
Isn't that solved in 2.0.3-post1?
